wandb/sdk/wandb_artifacts.py

```diff
@@ -91,6 +92,7 @@ class Artifact(ArtifactInterface):
         type: (str) The type of the artifact, which is used to organize and differentiate
             artifacts. Common types include `dataset` or `model`, but you can use any string
             containing letters, numbers, underscores, hyphens, and dots.
+        aliases: (list, optional) The list of aliases to associate with this artifact.
```

Comment:
To which version of the artifact do these aliases get associated?

As a regular user, it's obvious to me that they should be associated with `v0`. Is that the behavior?

Reply:
They would be associated with whatever version this artifact becomes.  If it happend to have the same contents as the current version, let's say `v4` then the aliases would be applied to it.  If the contents change when we log it the version would be bumped and the aliases associated with the new version.

Reply:
@annirudh I wonder if we should still associate the latest alias in this case?  The only scenario I can think of when a user wouldn't want the latest artifact to be marked as latest is if they want to record a "draft" artifact.  Maybe we could add a special alias like "draft" that prevented the auto "latest" logic...

Reply:
Echoing what @vanpelt said, this `aliases` list is what will be associated with the artifact that is being logged. If that list is `['production']`, then the version being logged will have the `production` alias. I think this is worth clarifying in the docstring.